packages/server/src/chatbot/chatbot.controller.ts

```diff
@@ -378,6 +384,103 @@ export class ChatbotController {
   // }
 
   // resets all chatbot data for the course. Unused
+
+  // Professor-only: send notification email to all students who asked this question
+  // Body must include oldAnswer and newAnswer, and can optionally include question changes for context
+  @Post('question/:courseId/:questionId/notify')
+  @UseGuards(CourseRolesGuard)
+  @Roles(Role.PROFESSOR)
+  async notifyUpdatedAnswer(
+    @Param('courseId', ParseIntPipe) courseId: number,
+    @Param('questionId') questionId: string,
+    @Body()
+    body: {
+      oldAnswer: string;
+      newAnswer: string;
+      oldQuestion?: string;
+      newQuestion?: string;
+    },
+    @User({ chat_token: true }) user: UserModel,
+  ): Promise<{ recipients: number }> {
+    handleChatbotTokenCheck(user);
+
+    const { oldAnswer, newAnswer, oldQuestion, newQuestion } = body ?? {};
+    if (oldAnswer == undefined || newAnswer == undefined) {
+      throw new BadRequestException(
+        'oldAnswer and newAnswer must be provided to send notifications.',
+      );
+    }
+    // Enforce: only allow notify when answer changed
+    if ((oldAnswer ?? '').trim() === (newAnswer ?? '').trim()) {
+      throw new BadRequestException(
+        'Cannot send notification: answer was not changed.',
+      );
+    }
+
+    // Find all interactions in this course where this question was asked
+    const asked = await ChatbotQuestionModel.find({
+      where: { vectorStoreId: questionId },
```

Comment:
```suggestion
      where: { 
        vectorStoreId: questionId
        course: {
          id: courseId 
      },
```

Saves needing to filter it later.